geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala

```diff
@@ -145,17 +145,17 @@ class StreamFeatureStore(entry: ContentEntry,
 }
 
 object StreamDataStoreParams {
-  val STREAM_DATASTORE_CONFIG = new Param("geomesa.stream.datastore.config", classOf[String], "", true)
-  val CACHE_TIMEOUT = new Param("geomesa.stream.datastore.cache.timeout", classOf[java.lang.Integer], "", true, 10)
+  val STREAM_DATASTORE_CONFIG = new GeoMesaParam[String]("geomesa.stream.datastore.config", "", required = true)
+  val CACHE_TIMEOUT = new GeoMesaParam[Integer]("geomesa.stream.datastore.cache.timeout", "", required = true, default = 10)
 }
 
 class StreamDataStoreFactory extends DataStoreFactorySpi {
 
   import StreamDataStoreParams._
 
   override def createDataStore(params: ju.Map[String, java.io.Serializable]): DataStore = {
-    val confString = STREAM_DATASTORE_CONFIG.lookUp(params).asInstanceOf[String]
-    val timeout = Option(CACHE_TIMEOUT.lookUp(params)).map(_.asInstanceOf[Int]).getOrElse(10)
+    val confString = STREAM_DATASTORE_CONFIG.lookup(params)
+    val timeout = CACHE_TIMEOUT.lookupOpt(params).map(_.intValue).getOrElse(10)
```

Comment:
While we are here....  should CACHE_TIMEOUT be all caps?

Reply:
I'll make it camel case

Reply:
fixed